### PR TITLE
Unrental pt2

### LINKED ISSUE #1024: Free operators from transmute
**Describe the problem you are trying to solve**

In the trickle operator, we use a transmute due to limitations on what we can express in the trait, perhaps there is a way around this.

```rust
            // This lifetimes will be `&'run mut Value<'event>` as that is the
            // requirement of the `self.runtime.run` we can not declare them
            // as the trait function for the operator doesn't allow that
```

This is based on the pipeline outliving the event.

**Notes**

Followup to #1015

### LINKED ISSUE #1037: replace rental
**Describe the problem you are trying to solve**

`rental` the library we are using for self-referential structs is no longer maintained and good alternatives like self_cell are now popping up. This is a good moment to consider replacing the rental code and freeing us from an unmaintained dependency.

**Describe the solution you'd like**

no more rental

**Notes**

This could be done as a multi-step project replacing one usage at a time.

### PR DESCRIPTION
**Pull request by Licenser**
# Pull request

## Description

Remove rental from everything but the Kafka source.

## Related

* Related Issues: fixes #1037, fixes #1024

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

## Performance

Slight, but repeated performance increase of 1-2%

### CODE REVIEW

File: tremor-cli/Cargo.toml [hunk 1]
```diff
@@ -35,31 +35,30 @@ log4rs = "1.0.0"
...
+tremor-pipeline = { path="../tremor-pipeline" }
+tremor-runtime = { path="../" }
+tremor-script = { path="../tremor-script" }
 url = "2"
 # mimalloc-rs = { version = "0.1", default-features = true, optional = true }
 # allocator_api = "0.6.0"
 error-chain = "0.12"
 globwalk = "0.8"
 port_scanner = "0.1"
-rental = "0.5"
```
Comment (mfelsche) [inline tremor-cli/Cargo.toml:59]: ❤️

File: tremor-script/src/srs.rs [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+            v: Value::from(v),
+            m: Value::from(m),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    // /// this is commented out since it should fail
```
Comment (mfelsche) [inline tremor-script/src/srs.rs:713]: you could move this to a doctest using `compile_fail` like so:

```rust
/// ```compile_fail
/// let x = 5;
/// x += 2; // shouldn't compile!
/// ```
```

File: tremor-value/src/value.rs [hunk 3]
```diff
@@ -91,9 +91,15 @@ impl<'value> Eq for Value<'value> {}
 impl<'value> Value<'value> {
     /// Creates an empty array value
     #[must_use]
-    pub fn array() -> Self {
+    pub const fn array() -> Self {
```
Comment (mfelsche) [inline tremor-value/src/value.rs:94]: ❤️

**Review by mfelsche — DISMISSED** [review]
LGTM!

That was a lot of reworking, but it turns out so much more elegant! Totally worth it!

File: tremor-script/src/srs.rs (hunk 2 above)
Comment (Licenser) [reply tremor-script/src/srs.rs:713]: I like that!

**Review by mfelsche — APPROVED** [review]
Looking even more GTM!

**Pull request merged**